app/frontend/src/cljs/teet/common/common_controller.cljs

```diff
@@ -373,25 +372,31 @@
 
 (def ^:private poll-timeout-ms 30000)
 
+(defonce version-seen-on-startup (atom nil))
 (defn poll-version
   "Polls whether the client's version of the app is the same as the one
   currently deployed. If not, shows a warning that a new version is
   being, or has been, deployed."
   [e!]
   (-> (fetch-deploy-status)
       (.then (fn [{:strs [commit status]}]
+               (when (and (some? commit)
+                          (nil? @version-seen-on-startup))
+                 (reset! version-seen-on-startup commit))
```

Comment:
Nitpicking, but you could just `(swap! version-seen-on-startup (fn [version] (or version commit)))` to avoid the extra deref.